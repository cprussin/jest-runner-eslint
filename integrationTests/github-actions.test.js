const { version } = require('jest/package.json');
const semver = require('semver');
const runJest = require('./runJest');

// NOTE: Jest versions <28 don't have a github action reporter, so this test is
// not valid.
if (semver.satisfies(version, '<28')) {
  it.skip('Reports with the github actions reporter', () => {});
} else {
  it('Reports with the github actions reporter', async () => {
    const priorActionsConfig = process.env.GITHUB_ACTIONS;
    process.env.GITHUB_ACTIONS = true;
    expect(await runJest('github-actions')).toMatchSnapshot();
    process.env.GITHUB_ACTIONS = priorActionsConfig;
  });
}
