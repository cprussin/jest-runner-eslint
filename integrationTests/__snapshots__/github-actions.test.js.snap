// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports with the github actions reporter 1`] = `
"
::error file=/mocked-path-to-jest-runner-mocha/integrationTests/__fixtures__/github-actions/__eslint__/failing.js,line=4,title=no-undef::'hello' is not defined.%0A%0A    at __eslint__/failing.js:4:3

::error file=/mocked-path-to-jest-runner-mocha/integrationTests/__fixtures__/github-actions/__eslint__/failing.js,line=5,title=no-undef::'bye' is not defined.%0A%0A    at __eslint__/failing.js:5:3
"
`;
